### ISSUE TITLE
support mycli access to vtgate

### ISSUE THREAD

**Issue by demmer** [vitessio/vitess#4365]
## Description
The mycli command line client complains with an angry stack trace when connecting to vtgate.

## Issue
There are certain queries that mycli runs at startup to get information about the databases which are incompletely supported by vtgate.

A quick scan through the source shows several of them here:
https://github.com/dbcli/mycli/blob/7047c56a0531cc5477543dec43305755d79216a6/mycli/sqlexecute.py#L23-L43

## Details
Running with version 1.5.2 of mycli against vtgate results in an error processing the initial `show databases` query:

```
Exception in thread completion_refresh:
Traceback (most recent call last):
  File "/usr/lib/python3.5/threading.py", line 914, in _bootstrap_inner
    self.run()
  File "/usr/lib/python3.5/threading.py", line 862, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/share/mycli/mycli/completion_refresher.py", line 59, in _bg_refresh
    refresher(completer, executor)
  File "/usr/share/mycli/mycli/completion_refresher.py", line 86, in refresh_databases
    completer.extend_database_names(executor.databases())
  File "/usr/share/mycli/mycli/sqlexecute.py", line 158, in databases
    cur.execute(self.databases_query)
  File "/usr/lib/python3/dist-packages/pymysql/cursors.py", line 158, in execute
    result = self._query(query)
  File "/usr/lib/python3/dist-packages/pymysql/cursors.py", line 308, in _query
    conn.query(q)
  File "/usr/lib/python3/dist-packages/pymysql/connections.py", line 820, in query
    self._affected_rows = self._read_query_result(unbuffered=unbuffered)
  File "/usr/lib/python3/dist-packages/pymysql/connections.py", line 1002, in _read_query_result
    result.read()
  File "/usr/lib/python3/dist-packages/pymysql/connections.py", line 1292, in read
    self._read_result_packet(first_packet)
  File "/usr/lib/python3/dist-packages/pymysql/connections.py", line 1346, in _read_result_packet
    self._get_descriptions()
  File "/usr/lib/python3/dist-packages/pymysql/connections.py", line 1416, in _get_descriptions
    charset = charset_by_id(field.charsetnr)
  File "/usr/lib/python3/dist-packages/pymysql/charset.py", line 34, in by_id
    return self._by_id[id]
KeyError: 0

Version: 1.5.2
Chat: https://gitter.im/dbcli/mycli
Mail: https://groups.google.com/forum/#!forum/mycli-users
Home: http://mycli.net
Thanks to the contributor - www.mysqlfanboy.com
percona god@localhost:(none)> show databases;
0
percona god@localhost:(none)> show databases;
Packet sequence number wrong - got 3 expected 1
percona god@localhost:(none)>
```

This is fixed by #4364, however even with that in place, a subsequent connection fails with:

```
Version: 1.5.2
Chat: https://gitter.im/dbcli/mycli
Mail: https://groups.google.com/forum/#!forum/mycli-users
Home: http://mycli.net
Exception in thread completion_refresh:
Traceback (most recent call last):
  File "/usr/lib/python3.5/threading.py", line 914, in _bootstrap_inner
    self.run()
  File "/usr/lib/python3.5/threading.py", line 862, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/share/mycli/mycli/completion_refresher.py", line 59, in _bg_refresh
    refresher(completer, executor)
  File "/usr/share/mycli/mycli/completion_refresher.py", line 102, in refresh_users
    completer.extend_users(executor.users())
  File "/usr/share/mycli/mycli/sqlcompleter.py", line 115, in extend_users
    for user in users:
  File "/usr/share/mycli/mycli/sqlexecute.py", line 186, in users
    cur.execute(self.users_query)
  File "/usr/lib/python3/dist-packages/pymysql/cursors.py", line 158, in execute
    result = self._query(query)
  File "/usr/lib/python3/dist-packages/pymysql/cursors.py", line 308, in _query
    conn.query(q)
  File "/usr/lib/python3/dist-packages/pymysql/connections.py", line 820, in query
    self._affected_rows = self._read_query_result(unbuffered=unbuffered)
  File "/usr/lib/python3/dist-packages/pymysql/connections.py", line 1002, in _read_query_result
    result.read()
  File "/usr/lib/python3/dist-packages/pymysql/connections.py", line 1285, in read
    first_packet = self.connection._read_packet()
  File "/usr/lib/python3/dist-packages/pymysql/connections.py", line 966, in _read_packet
    packet.check_error()
  File "/usr/lib/python3/dist-packages/pymysql/connections.py", line 394, in check_error
    err.raise_mysql_exception(self._data)
  File "/usr/lib/python3/dist-packages/pymysql/err.py", line 120, in raise_mysql_exception
    _check_mysql_exception(errinfo)
  File "/usr/lib/python3/dist-packages/pymysql/err.py", line 115, in _check_mysql_exception
    raise InternalError(errno, errorvalue)
pymysql.err.InternalError: (1105, 'vtgate: http://slack-vtgate-dev-loadtest-fghj.ec2.tinyspeck.com:15000/: no keyspace in database name specified. Supported database name format (items in <> are optional): keyspace<:shard><@type> or keyspace<[range]><@type>')
```

**Comment by deepthi** [comment]
can you tell from vtgate_querylog whether there is a specific query that is causing the error?

**Comment by harshit-gangal** [comment]
Fixed via #6667
